components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/APIManagerTenantSyncListener.java

```diff
@@ -0,0 +1,371 @@
+    }
+
+    @Override
+    public void onPreDelete(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    /**
+     * Check whether tenant sharing is enabled and whether the creation is only a root organization.
+     * 
+     * @param tenantId id of the tenant
+     * @return boolean
+     */
+    private boolean canCreateTenantInAPIM(int tenantId) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Checking if tenant can be created in API Manager. Tenant ID: " + tenantId);
+        }
+
+        boolean canCreateTenant = false;
+
+        IdentityEventListenerConfig identityEventListenerConfig = IdentityUtil.readEventListenerProperty(
+                UserOperationEventListener.class.getName(), APIManagerTenantSyncListener.class.getName());
+
+        if (StringUtils.isNotBlank(identityEventListenerConfig.getEnable())
+                && Boolean.parseBoolean(identityEventListenerConfig.getEnable())) {
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+
+                Tenant tenant = realmService.getTenantManager().getTenant(tenantId);
+                String organizationID = tenant.getAssociatedOrganizationUUID();
+
+                // check if the Organization Depth in the Hierarchy is 0. only then create the root org.
+                if (organizationID == null || IdentityMgtServiceDataHolder.getInstance().getOrganizationManager()
+                        .getOrganizationDepthInHierarchy(organizationID) == 0) {
+
+                    canCreateTenant = true;
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Tenant is a root organization and can be created in API Manager. Tenant ID: "
+                                + tenantId);
+                    }
+
+                } else {
+                    log.debug("Skipping creating the tenant in APIM since the triggered Event is not related "
+                            + "to a root org creation.");
+                }
+
+                // if there was an exception thrown here, tenant activation won't happen
+            } catch (UserStoreException | OrganizationManagementServerException e) {
+                log.error("Error while creating tenant in API Manager ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        } else {
+            log.debug(
+                    "Tenant sharing is disabled. Skipping tenant creation in API Manager for tenant ID : " + tenantId);
+        }
+        return canCreateTenant;
+    }
+
+    /**
+     * Method to build the payload and send it to the external server. Event is sent asynchronously
+     * 
+     * @param tenantInfo
+     * @param type
+     * @param eventURI
```

Comment:
complete the comment

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/a6be893041c5d4a898b3d2f74d8525409740cf5c